src/Compilers/VisualBasic/Portable/Symbols/Attributes/AttributeData.vb

```diff
@@ -271,10 +288,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                      DeclarativeSecurityAction.LinkDemand
 
                     If Me.IsTargetAttribute(targetSymbol, AttributeDescription.PrincipalPermissionAttribute) Then
-                        ' BC31209: SecurityAction value '{0}' is invalid for PrincipalPermission attribute
-                        diagnostics.Add(ERRID.ERR_PrincipalPermissionInvalidAction,
-                                        If(nodeOpt IsNot Nothing, nodeOpt.ArgumentList.Arguments(0).GetLocation(), NoLocation.Singleton),
-                                        If(nodeOpt IsNot Nothing, nodeOpt.ArgumentList.Arguments(0).ToString(), ""))
+                        ' BC31215: SecurityAction value '{0}' is invalid for PrincipalPermission attribute
+                        Dim valueLocation = GetArgumentAndLocation(nodeOpt, securityAction)
```

Comment:
>Dim [](http://example.com/codeflow?start=24&length=3)

Please spell out the type. We are not using implicit typing in declarations when the type isn't obvious from the initial value (i.e. not somehow spelled out there).